states/texas/elections.toml

```diff
@@ -1,13 +1,46 @@
-[20200303]
-date = 2020-03-03
-name = "Congressional & Presidential Primary"
-original_date = 2020-03-03
-
 [20200714]
 date = 2020-07-14
 name = "Congressional Primary Run-Off (if needed)"
 original_date = 2020-07-14
 
+[20200714.registration]
+received_by = 2020-06-15
+postmarked_by = 2020-06-15
+in_person_by = 2020-06-15
+sources = ["https://www.sos.state.tx.us/elections/voter/important-election-dates.shtml",
+            "https://www.votetexas.gov/register-to-vote/you-must-register-by.html",
+            "https://www.votetexas.gov/faq/index.html#2798"]
+
+[20200714.poll]
+in_person_starts = 2020-07-14T07:00:00
+in_person_by = 2020-07-14T19:00:00
+sources = ["https://www.sos.state.tx.us/elections/voter/important-election-dates.shtml"]
+
+[20200714.poll.early]
+in_person_starts = 2020-06-29
+in_person_by = 2020-07-10
+sources = ["https://www.sos.state.tx.us/elections/voter/important-election-dates.shtml",
+           "https://www.votetexas.gov/faq/early-voting.html#2773"]
+
+[20200714.absentee]
+received_by = 2020-07-15T17:00:00
+postmarked_by = 2020-07-14T19:00:00
+sources = ["https://www.sos.state.tx.us/elections/voter/important-election-dates.shtml",
+           "https://www.votetexas.gov/military-overseas-voters/index.html",
+           "https://www.sos.texas.gov/elections/voter/reqabbm.shtml"]
+
+[20200714.absentee.application]
+received_by = 2020-07-02
+sources = ["https://www.sos.state.tx.us/elections/voter/important-election-dates.shtml",
+           "https://www.votetexas.gov/military-overseas-voters/index.html",
+           "https://www.sos.texas.gov/elections/voter/reqabbm.shtml"]
+
+
+[20200714.absentee.overseas_military]
+received_by = 2020-07-21
```

Comment:
So this is interesting, the overseas and military are 1 day apart. How would we handle it?

![image](https://user-images.githubusercontent.com/208567/85937671-16ddf580-b8c3-11ea-9824-115a320161f1.png)

Reply:
Could just do them as separate items. @askpatrickw what do you think?

Reply:
The more I see these, I agree they should be broken out.

That would let us have different feeds tailored to specific kinds of voters in the future, and other kinds of flexibility.

Where available we can enter these dates, but I was also thinking once all states are entered, I would go back through and check the federal site for this (FVAP I think) and double check that site with the state sites and the TOML.

Reply:
@askpatrickw @tannewt I'm about to go get coffee and fix a coffee grinder.  When I get back I can make changes depending on what y'all decide!

I know in Colorado mail in voting is a default so your ballot gets mailed to you automatically, but I do believe it's different that absentee since that is just for people out of state/country (mostly military, but also expats to an extent). There are different deadlines for mail in ballots than dropping the ballot off in person, the latter of which usually coincides with the actual election day. I think it would be more clear to specify in state vs out of state mail in voting for those who have the option. Whether this should be a new keyword or a new field, I am unsure!

Reply:
Would the out of state case be covered by `overseas` and `military`? Seems like it would be.

Reply:
Well most of the time yes, but in the case of texas, overseas military have a day extra vs overseas expats which is also different from the mail in voting for instate people, people with disabilities, over 65, or people who are incarcerated but can still vote.

Reply:
Ok, I think that is covered with `absentee`, `overseas` and `military`. Make sense?

Reply:
What do you think about this?

```toml
[yymmdd.registration]     # Normal registration rules

[yyyymmdd.absentee.application]     # Send me my mail in ballot some place else in the U.S.
[yyyymmdd.absentee.application.overseas]     #Send me my mail in ballot to another country # I bet this was added for other foreign service originally
[yyyymmdd.absentee.application.military]     # Special Military absentee rules 

[yyyymmdd.absentee]     # deadlines for u.s. absentee 
[yyyymmdd.absentee.overseas]     # deadlines for overseas absentee 
[yyyymmdd.absentee.military]     # deadlines for military absentee 
```

There many also need to be a `[yyyymmdd.absentee.application.emergency]`

Reply:
This looks good to me!

Reply:
@tannewt @askpatrickw I updated the texas primary to abide by the proposed schema above by Patrick. If they are all the same for the application do we still need to split them out or is there existing code that will populate them based on the "deafult".